### PR TITLE
Fix connection issues

### LINKED ISSUE #1: Cannot bind 5353 port in linux
UDP 5353 port is used by zeroconf service, rust can’t bind it again.

however, golang ListenUDP works fine.

### PR DESCRIPTION
**Pull request by metaclips**
closes #1

### CODE REVIEW

**Comment by metaclips** [comment]
https://github.com/webrtc-rs/mdns/pull/3/commits/85fdfc9ad7e588993f74fb0c8a9e37abdccea5e4 this was me debugging 🙁🙁 Apologies

File: src/conn/mod.rs [hunk 1]
```diff
@@ -179,14 +183,14 @@ impl DNSConn {
         let raw_query = match msg.pack() {
             Ok(v) => v,
             Err(err) => {
-                warn!("Failed to construct mDNS packet {}", err);
+                println!("Failed to construct mDNS packet {}", err);
```
Comment (rainliu) [inline src/conn/mod.rs:186]: use log::warn?

File: src/conn/mod.rs [hunk 2]
```diff
@@ -16,11 +14,9 @@ use tokio::sync::Mutex;
 
 use util::Error;
 
-use log::*;
-
 const INBOUND_BUFFER_SIZE: usize = 512;
 const DEFAULT_QUERY_INTERVAL: Duration = Duration::from_secs(1);
-const DEFAULT_DEST_ADDR: &str = "224.0.0.251";
+const DEFAULT_DEST_ADDR: &str = "224.0.0.251:5353";
 const DEFAULT_DEST_PORT: u16 = 5353;
```
Comment (rainliu) [inline src/conn/mod.rs:20]: remove unused const

File: Cargo.toml [hunk 3]
```diff
@@ -16,12 +16,13 @@ util = { package = "webrtc-rs-util", version = "0.1.2" }
 lazy_static = "1.3.0"
 tokio = { version = "1.0", features = ["full"] }
 log = "0.4"
-
+net2 = "*"
```
Comment (rainliu) [inline Cargo.toml:19]: looks like net2 is Deprecated, and  are encouraged to consider socket2 instead, according to https://crates.io/crates/net2

Comment (metaclips) [reply Cargo.toml:19]: Thanks. Will implement changes.

**Comment by metaclips** [comment]
@rainliu @Sean-Der there seem to be a failing test :( sockets fails to receive packets and I really don't know why as it works fine on the server/query example. An extra eye will be fantastic.

**Comment by metaclips** [comment]
Fixed all tests do review @rainliu @Sean-Der

File: src/conn/mod.rs [hunk 4]
```diff
@@ -179,53 +222,65 @@ impl DNSConn {
...
         queries: Arc<Mutex<Vec<Query>>>,
     ) -> Result<(), Error> {
-        let mut b = vec![0u8; INBOUND_BUFFER_SIZE];
-
-        let (mut n, mut src);
+        log::info!("enter loop and listening {:?}", socket.local_addr());
 
         loop {
-            trace!("enter loop and listening {:?}", socket.local_addr());
+            let mut b = vec![0u8; INBOUND_BUFFER_SIZE];
```
Comment (rainliu) [inline src/conn/mod.rs:247]: Will there be any performance issue to allocate vector b inside loop ? Is this change necessary?

Comment (metaclips) [reply src/conn/mod.rs]: None whatsoever. Fixed changes.

**Pull request merged**